ci/do_ci.sh

```diff
@@ -23,37 +23,101 @@ function do_coverage() {
     ci/run_coverage.sh
 }
 
-CONCURRENCY=8
+function setup_gcc_toolchain() {
+    export CC=gcc
+    export CXX=g++
+    echo "$CC/$CXX toolchain configured"
+}
+
+function setup_clang_toolchain() {
+    export PATH=/usr/lib/llvm-7/bin:$PATH
+    export CC=clang
+    export CXX=clang++
+    export ASAN_SYMBOLIZER_PATH=/usr/lib/llvm-7/bin/llvm-symbolizer
+    echo "$CC/$CXX toolchain configured"
+}
+
+# This overrides both ld and ld.gold with llvm's ld.lld.
```

Comment:
This is an insane (in both the good and bad way ;) ) hack; kudos for figuring out a way to get this to work. OTOH, I think this should be tagged with a big TODO and a tracking issue, at some point we want to unwind this and figure out why this isn't working as expected.

Reply:
Done